### ISSUE TITLE
[HOLD for payment 2025-01-07] Web - LHN - Workspace switcher resets to Expensify when sending invoice for the first time

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#54509]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.78-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** No
**If this was caught during regression testing, add the test name, ID and link from TestRail:** Exp
**Email or phone of affected tester (no customers):** applausetester+909608@applause.expensifail.com
**Issue reported by:** Applause Internal Team
**Device used:** Mac 15.0 / Chrome
**App Component:** Left Hand Navigation (LHN)

## Action Performed:
1. Go to staging.new.expensify.com
2. Create a new workspace.
3. Go to workspace settings > More features.
4. Enable Invoices.
5. Go back to Inbox.
6. Open workspace switcher,
7. Select the new workspace.
8. Open FAB > Send invoice.
9. Send invoice from the new workspace.
10. Note that workspace switcher resets to Expensify after sending invoice.
11. Repeat Step 8 to 9.
12. Note that workspace switcher does not reset to Expensify when sending invoice for the second time.

## Expected Result:
In Step 10, the workspace switcher will not reset to Expensify when sending invoice for the first time (production behavior).

## Actual Result:
In Step 10, the workspace switcher resets to Expensify when sending invoice for the first time.

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/8ca041e6-ede7-4e90-97c7-e21a477f3d1d

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @laurenreidexpensify</details>

**Comment by cristipaval** [comment]
> ~It might be expected behaviour given that you are sending invoice to chat outside the workspace.~
> 
> OK, so this is a real issue. We are sending invoice on workspace chat.

DANG! This is bad!

**Comment by cristipaval** [comment]
> Looking for issue.

Thanks @parasharrajat!

**Comment by jasperhuangg** [comment]
I don't think we need to block deploy on this, but @chrispader please look into this when you can. https://github.com/Expensify/App/pull/54030 looks a bit too complicated to revert.

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
WS switcher resets to all when sending invoice.

### What is the root cause of that problem?
When we send a new invoice, we will switch to the policy that we use to send the invoice. Before https://github.com/Expensify/App/pull/54030, when we navigate to a HOME screen and the home screen already exists on the bottom tab stack, it will simply update the params, but now, it will always push a new screen.
<img width="1106" alt="image" src="https://github.com/user-attachments/assets/bfe11273-e0ac-4d14-8293-13c1d7c66df2" />

And a new sidebar screen is mounted. In the sidebar screen, we have an effect. that the purpose is to switch to the all workspace view if the active workspace is not exist.
https://github.com/Expensify/App/blob/ad99c20c40e4fed347cff29f5ed85c98825442a8/src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.tsx#L24-L38

In our case, the activeWorkspaceID is available, but the activeWorkspace object is not, and that is because it's still loading.
https://github.com/Expensify/App/blob/ad99c20c40e4fed347cff29f5ed85c98825442a8/src/libs/actions/IOU.ts#L1417-L1431

Why is it loading only when we send an invoice to a new workspace? It's because when we send an invoice to a new workspace, we submit the company name and website. So, the onyx has a pending merge and `useOnyx` will return it as a loading.
https://github.com/Expensify/react-native-onyx/blob/620fccfc9c3dd7704526f1a453a39f46de6d956d/lib/useOnyx.ts#L245-L248

### What changes do you think we should make in order to solve the problem?

Return early when the onyx is still loading.
```
const [activeWorkspace, activeWorkspaceResult] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY}${activeWorkspaceID ?? CONST.DEFAULT_NUMBER_ID}`);
const isLoading = isLoadingOnyxValue(activeWorkspaceResult);

useEffect(() => {
    if (!!activeWorkspace || activeWorkspaceID === undefined || isLoading) {
        return;
    }

    Navigation.navigateWithSwitchPolicyID({policyID: undefined});
    updateLastAccessedWorkspace(undefined);
}, [activeWorkspace, activeWorkspaceID, isLoading]);
```

(or we can set `allowStaleData` to true for the useOnyx, but I prefer the above)

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?

N/A

**Comment by bernhardoj** [comment]
In case we want to make it external, I have a proposal above.

**Comment by chrispader** [comment]
> I don't think we need to block deploy on this, but @chrispader please look into this when you can. #54030 looks a bit too complicated to revert.

This could have either been caused by any changes made between the causing PR and the NativeStack PR where the reverted change was introduced.

One change that could have affected this, was the change of the fallback of the `activeWorkspaceID` from `-1` to `CONST.DEFAULT_NUMBER_ID`.

The proposal above looks good to me!

**Comment by bernhardoj** [comment]
PR is ready

cc: @parasharrajat

**Comment by laurenreidexpensify** [comment]
- Reviewer: @parasharrajat owed $250 via NewDot
- Reviewer: @bernhardoj owed $250 via NewDot

Please request payment in ND thanks

**Comment by bernhardoj** [comment]
Requested in ND.

**Comment by JmillsExpensify** [comment]
$250 approved for @bernhardoj

**Comment by JmillsExpensify** [comment]
$250 approved for @parasharrajat
